dbms/src/TiDB/Schema/TiDB.cpp

```diff
@@ -452,6 +466,27 @@ try
         collate = type_json->get("Collate");
     state = static_cast<SchemaState>(json->getValue<Int32>("state"));
     comment = json->getValue<String>("comment");
+
+    auto vector_index_json = json->getObject("vector_index");
+    if (vector_index_json)
+    {
+        vector_index = std::make_shared<VectorIndexInfo>();
+
+        auto vector_kind = magic_enum::enum_cast<VectorIndexKind>(vector_index_json->getValue<String>("kind"));
+        RUNTIME_CHECK(vector_kind.has_value());
+        RUNTIME_CHECK(vector_kind.value() != VectorIndexKind::INVALID);
+        vector_index->kind = vector_kind.value();
+
+        vector_index->dimension = vector_index_json->getValue<UInt64>("dimension");
+        RUNTIME_CHECK(vector_index->dimension > 0);
+        RUNTIME_CHECK(vector_index->dimension <= 16000); // Just a protection
```

Comment:
```suggestion
        RUNTIME_CHECK(vector_index->dimension <= 16383); // Just a protection
```

This should be updated to 16383 for MySQL compatibility according to the feature spec https://github.com/pingcap/tidb/pull/55002#discussion_r1703687255